lib/puppetserver/ca/action/list.rb

```diff
@@ -165,19 +211,28 @@ def separate_certs(all_certs)
 
         def get_all_certs(settings)
           result = Puppetserver::Ca::CertificateAuthority.new(@logger, settings).get_certificate_statuses
-          result ? JSON.parse(result.body) : []
+
+          if result
+            return JSON.parse(result.body)
+          else 
+            return []
+          end
         end
 
         def parse(args)
           results = {}
           parser = self.class.parser(results)
+          exit_code = 0
```

Comment:
why are you setting this here?

Reply:
Declare a variable before setting it? Or should I have just return the exit code inside of the if/else block. My old habit of writing C code is showing...

Reply:
Yeah  ( : 
You don't have to declare a variable or zero it out before using it.
This is just setting it to one thing and then mutating it later on, you can remove this line and then leave the rest as-is.

Reply:
Ruby's scopes are interesting. Didn't know I could use a variable declared inside an if statement to be usable outside of it. Pretty neat.